Libraries/CMSIS/Device/Maxim/MAX32657/Source/GCC/setup_memory_tz.py

```diff
@@ -0,0 +1,731 @@
+                        # Check MPC alignment.
+                        block, remainder = divmod(self.phy_inst_sizes[curr_region_start_mem_inst] - curr_start, self.mpc_inst_blk_sizes[curr_region_start_mem_inst])
+
+                        if remainder != 0:
+                            MPC_BLK_SIZE = integer_bytes_to_string_KiB(self.mpc_inst_blk_sizes[curr_region_start_mem_inst])
+                            print(f"> [ERROR]: Incorrect '{self.type}' Memory Region Alignment. The memory regions must be aligned with the corresponding '{self.type}' Memory Protection Controller (MPC) Block Size: '{MPC_BLK_SIZE}'")
+                            sys.exit(1)
+
+                # Save current region before proceeding to next iteration.
+                prev_start = curr_start
+                prev_size = curr_size
+
+    # Set up Security regions.
+    def setRegions(self, secure_start, secure_size,
+                    nonsecure_start, nonsecure_size,
+                    nonsecurecall_start, nonsecurecall_size, is_nonsecurecall_present):
+
+        # Constructor only checks for physical memory constraints.
+        #   This checks for user configured memory settings.
+        starting_region_start = min(secure_start & ~(1<<28), nonsecure_start)
+        if (starting_region_start) != self.phy_start:
+            # Save bit 28 state for error message
+            if (secure_start & ~(1<<28)) == starting_region_start:
+                bit28_security = (1 << 28)
+            else:
+                bit28_security = 0
+
+            print(f"> [ERROR]: Lowest Memory Region does not start at the beginning of '{self.type}': '{hex(self.phy_start | bit28_security)}'")
+            sys.exit(1)
+
+        self.s_start = secure_start
+        self.s_size = secure_size
+        self.ns_start = nonsecure_start
+        self.ns_size = nonsecure_size
+
+        self.are_regions_set = True
+        self.is_nonsecurecall_present = is_nonsecurecall_present
+
+        # The NSC region is a subset of the Secure region. Treat it holistically - if NSC is present in this memory type.
+        self.total_s_nsc_size = secure_size
+
+        # Leave as default (None) set by constructor if NSC region not present in this memory type.
+        if is_nonsecurecall_present:
+            self.nsc_start = nonsecurecall_start
+            self.nsc_size = nonsecurecall_size
+            self.total_s_size = secure_size + nonsecurecall_size
+        
+        self.__check_region_alignment_to_mpc()
+
+
+###############################################################################
+#
+# Instantiate the argument parser.
+#
+
+program_description = '''Generates the linker scripts and partiton_{device}.h with
+project-defined memory settings. 
+
+NOTE: This script uses binary prefix notation for units when not used by build system
+For example: Kilobytes (KB) will be treated as Kibibytes (KiB).
+
+Note: All these arguments will be set with default values according to the {device}.mk file.
```

Comment:
Only because the build system calls this script though, right?  This could be misleading since it doesn't parse the makefile